File: app/src/main/java/com/joker/maindexkeep/annotations/Type.java
package com.joker.maindexkeep.annotations;

import android.support.annotation.StringDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@RuntimeAnn
@Retention(RetentionPolicy.RUNTIME)
@StringDef({ Type.A, Type.B })
public @interface Type {
  String A = "a";
  String B = "b";
}

File: app/src/main/java/com/joker/maindexkeep/annotations/Type.kt
package com.joker.maindexkeep.annotations

import android.support.annotation.StringDef
import kotlin.annotation.AnnotationRetention.RUNTIME

@RuntimeAnn
@kotlin.annotation.Retention(RUNTIME)
@StringDef(Type.A, Type.B)
annotation class Type {
  companion object {
    const val A = "a"
    const val B = "b"
  }
}
